File: clinical-dashboard/get-data-refresh-date.js
const fs = require('fs');
const path = require('path');

// Directory containing data files
const dataDir = path.join(__dirname, 'data');

// Get all CSV files in the data directory
function getCSVFiles(dir) {
  const files = fs.readdirSync(dir);
  return files.filter(file => {
    const ext = path.extname(file).toLowerCase();
    return ext === '.csv' || (ext === '' && file.toUpperCase().endsWith('.CSV'));
  });
}

// Get the most recent modification date from all CSV files
function getMostRecentDate() {
  try {
    const csvFiles = getCSVFiles(dataDir);
    let mostRecentDate = null;

    csvFiles.forEach(file => {
      const filePath = path.join(dataDir, file);
      try {
        const stats = fs.statSync(filePath);
        const modifiedDate = stats.mtime;
        
        if (!mostRecentDate || modifiedDate > mostRecentDate) {
          mostRecentDate = modifiedDate;
        }
      } catch (err) {
        console.error(`Error reading ${file}:`, err.message);
      }
    });

    return mostRecentDate;
  } catch (err) {
    console.error('Error reading data directory:', err.message);
    return null;
  }
}

// Generate JSON file with the refresh date
function generateRefreshDateFile() {
  const mostRecentDate = getMostRecentDate();
  
  if (mostRecentDate) {
    const dateInfo = {
      refreshDate: mostRecentDate.toISOString(),
      formattedDate: mostRecentDate.toLocaleDateString('en-US', {
        year: 'numeric',
        month: 'numeric',
        day: 'numeric'
      }),
      timestamp: mostRecentDate.getTime()
    };

    const outputPath = path.join(__dirname, 'data', 'refresh-date.json');
    fs.writeFileSync(outputPath, JSON.stringify(dateInfo, null, 2));
    console.log(`Data refresh date updated: ${dateInfo.formattedDate}`);
    console.log(`File saved to: ${outputPath}`);
  } else {
    console.error('Could not determine data refresh date');
  }
}

// Run the script
generateRefreshDateFile();

File: clinical-dashboard/get_data_refresh_date.py
import os
import json
from pathlib import Path
from datetime import datetime

# Directory containing data files
data_dir = Path(__file__).parent / 'data'

# Get all CSV files in the data directory
def get_csv_files(directory):
    csv_files = []
    for file in directory.iterdir():
        if file.is_file():
            ext = file.suffix.lower()
            if ext == '.csv' or (ext == '' and file.name.upper().endswith('.CSV')):
                csv_files.append(file)
    return csv_files

# Get the most recent modification date from all CSV files
def get_most_recent_date():
    try:
        csv_files = get_csv_files(data_dir)
        most_recent_date = None

        for file in csv_files:
            try:
                modified_time = file.stat().st_mtime
                modified_date = datetime.fromtimestamp(modified_time)
                
                if most_recent_date is None or modified_date > most_recent_date:
                    most_recent_date = modified_date
            except Exception as e:
                print(f'Error reading {file.name}: {e}')

        return most_recent_date
    except Exception as e:
        print(f'Error reading data directory: {e}')
        return None

# Generate JSON file with the refresh date
def generate_refresh_date_file():
    most_recent_date = get_most_recent_date()
    
    if most_recent_date:
        date_info = {
            'refreshDate': most_recent_date.isoformat(),
            'formattedDate': most_recent_date.strftime('%m/%d/%Y'),
            'timestamp': int(most_recent_date.timestamp() * 1000)
        }

        output_path = data_dir / 'refresh-date.json'
        with open(output_path, 'w', encoding='utf-8') as f:
            json.dump(date_info, f, indent=2)
        print(f'Data refresh date updated: {date_info["formattedDate"]}')
        print(f'File saved to: {output_path}')
    else:
        print('Could not determine data refresh date')

# Run the script
if __name__ == '__main__':
    generate_refresh_date_file()

